### PR TITLE
Support href and cursor properties for marks

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/vega.schema.ts [hunk 1]
```diff
@@ -1056,6 +1056,18 @@ export interface VgMarkConfig {
...
+   * A URL to load upon mouse click. If defined, the mark acts as a hyperlink.
+   *
+   * @format uri
+   */
+  href?: string;
+
+  /**
+   * The mouse cursor used over the mark. Any valid [CSS cursor type](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor#Values) can be used.
+   */
+  cursor?: string;
```
Comment (kanitw) [inline src/vega.schema.ts]: The link you provided lists all cursor type.  Perhaps, make it an enum here?

auto | default | none | context-menu | help | pointer | progress | wait | cell | crosshair | text | vertical-text | alias | copy | move | no-drop | not-allowed | e-resize | n-resize | ne-resize | nw-resize | s-resize | se-resize | sw-resize | w-resize | ew-resize | ns-resize | nesw-resize | nwse-resize | col-resize | row-resize | all-scroll | zoom-in | zoom-out | grab | grabbing

File: src/vega.schema.ts [hunk 2]
```diff
@@ -1084,10 +1096,11 @@ const VG_MARK_CONFIG_INDEX: Flag<keyof VgMarkConfig> = {
   font: 1,
   fontSize: 1,
   fontWeight: 1,
-  fontStyle: 1
+  fontStyle: 1,
+  href: 1,
+  cursor: 1,
```
Comment (kanitw) [inline src/vega.schema.ts]: I'm not sure why previously list `cursor` as "vg channel that do not have mark config".  Can you test if cursor works below for sure?

Btw, does it really make sense to have `href` as a mark config too?

File: site/docs/mark/mark.md [hunk 3]
```diff
@@ -125,11 +125,16 @@ The rest of this section describe groups of properties supported by the `mark` c
...
 {% include table.html props="opacity,fillOpacity,strokeOpacity" source="MarkConfig" %}
 
-
 ### Stroke Style
 
 {% include table.html props="strokeWidth,strokeDash,strokeDashOffset" source="MarkConfig" %}
 
+### Link Properties
+
+Marks can act as hyperlinks when the `href` property is defined.
```
Comment (kanitw) [inline site/docs/mark/mark.md]: - `href` property -> `href` channel

- Maybe also add:

A `cursor` channel can also be provided to serve as affordance for the links.

Comment (domoritz) [reply site/docs/mark/mark.md]: Done

File: src/vega.schema.ts (hunk 1 above)
Comment (domoritz) [reply src/vega.schema.ts]: Okay. I don't know whether browser vendors will add more in the future, though.

File: src/vega.schema.ts (hunk 2 above)
Comment (domoritz) [reply src/vega.schema.ts]: Working on it

**Comment by domoritz** [comment]
I will finish this in a few days. Thanks for the review.

File: src/vega.schema.ts (hunk 2 above)
Comment (domoritz) [reply src/vega.schema.ts]:
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A","b": 28}, {"a": "B","b": 55}, {"a": "C","b": 43},
      {"a": "D","b": 91}, {"a": "E","b": 81}, {"a": "F","b": 53},
      {"a": "G","b": 19}, {"a": "H","b": 87}, {"a": "I","b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal"},
    "y": {"field": "b", "type": "quantitative"}
  },
  "config": {
    "bar": {
      "cursor": "pointer"
    }
  }
}
```

works

**Comment by domoritz** [comment]
Ready

File: site/docs/mark/mark.md [hunk 4]
```diff
@@ -125,11 +125,16 @@ The rest of this section describe groups of properties supported by the `mark` c
...
 {% include table.html props="opacity,fillOpacity,strokeOpacity" source="MarkConfig" %}
 
-
 ### Stroke Style
 
 {% include table.html props="strokeWidth,strokeDash,strokeDashOffset" source="MarkConfig" %}
 
+### Link Properties
+
+Marks can act as hyperlinks when the `href` channel is defined. A `cursor` channel can also be provided to serve as affordance for the links.
```
Comment (kanitw) [inline site/docs/mark/mark.md]: If you don't add them as channels, why the docs say that they are channels here?

Comment (domoritz) [reply site/docs/mark/mark.md]: Ups.

File: site/docs/encoding.md [hunk 5]
```diff
@@ -180,27 +180,27 @@ In addition to the constant `value`, [value definitions](#value-def) of mark pro
 See [the `condition`](condition.html) page for examples how to specify condition logic.
 
 {:#text}
-## Text and Tooltip Channels
+## Text, Tooltip, and HREF Channels
```
Comment (kanitw) [inline site/docs/encoding.md]: href should be on its own category -- it doesn't have `format` property

File: src/encoding.ts [hunk 6]
```diff
@@ -98,6 +98,11 @@ export interface Encoding<F> {
    */
   tooltip?: FieldDefWithCondition<TextFieldDef<F>> | ValueDefWithCondition<TextFieldDef<F>>;
 
+  /**
+   * A URL to load upon mouse click.
+   */
+  href?: FieldDefWithCondition<TextFieldDef<F>> | ValueDefWithCondition<TextFieldDef<F>>;
```
Comment (kanitw) [inline src/encoding.ts]: It shouldn't be TextFieldDef?

File: site/docs/encoding.md (hunk 5 above)
Comment (domoritz) [reply site/docs/encoding.md]: Hmm, I was thinking that one might want to format a number to be a URL like `www.site.com?page=${id}`.

File: src/encoding.ts (hunk 6 above)
Comment (domoritz) [reply src/encoding.ts]: What if you want a conditional url?

Comment (kanitw) [reply src/encoding.ts]: Well, TextFieldDef and conditional are orthogonal.  TextFieldDef has `format` that href shouldn't

File: site/docs/encoding.md (hunk 5 above)
Comment (kanitw) [reply site/docs/encoding.md]: Yeah but having `format` wouldn't help you do that anyway?

https://github.com/d3/d3-format#locale_format

Comment (kanitw) [reply site/docs/encoding.md]: oh I see -- you can use "fill"

Comment (kanitw) [reply site/docs/encoding.md]: Actually you can't .. fill is for a slightly different purpose

Comment (domoritz) [reply site/docs/encoding.md]: You're right. I was thinking of dates where you can do something like `foo %H bar`

File: src/encoding.ts (hunk 6 above)
Comment (domoritz) [reply src/encoding.ts]: Fixed and rebased.

File: site/docs/encoding.md [hunk 7]
```diff
@@ -204,6 +208,29 @@ In addition to the constant `value`, [value definitions](#value-def) of `text` a
 
 {% include table.html props="condition" source="TextValueDefWithCondition" %}
 
+{:#href}
+## Hyperlink Channel
+
+By setting the `href` channel, a mark becomes a hyperlink. The specified URL is loaded upon a muse click. The `cursor` mark property can be set to `pointer` to serve as affordance for hyperlinks.
```
Comment (kanitw) [inline site/docs/encoding.md]: add a link for "The `cursor` mark property"?

File: test/compile/mark/mark.test.ts [hunk 8]
```diff
@@ -120,6 +120,21 @@ describe('Mark', function() {
         assert.equal(markGroup[0].encode.update.tooltip.value, 'foo');
       });
     });
+
+    describe('Bar with href', () => {
```
[the remark this answers is not in the document]
Comment (domoritz) [reply test/compile/mark/mark.test.ts]: Okay. I'm moving the tooltip and href tests.

**Comment by domoritz** [comment]
Done with revisions

File: src/channel.ts [hunk 9]
```diff
@@ -128,7 +128,8 @@ export type PositionScaleChannel = typeof POSITION_SCALE_CHANNELS[0];
 // NON_POSITION_SCALE_CHANNEL = SCALE_CHANNELS without X, Y
 const {
   // x2 and y2 share the same scale as x and y
-  // text, tooltip, and href have format instead of scale
+  // text and tooltip have format instead of scale,
+  // href has neother format, nor scale
```
Comment (kanitw) [inline src/channel.ts]: neither

**Comment by domoritz** [comment]
cc @rdeline

**Pull request merged**